central/namespace/resolver.go

```diff
@@ -17,16 +18,28 @@ import (
 // ResolveAll resolves all namespaces, populating volatile runtime data (like deployment and secret counts) by querying related stores.
 func ResolveAll(ctx context.Context, dataStore datastore.DataStore, deploymentDataStore deploymentDataStore.DataStore,
 	secretDataStore secretDataStore.DataStore, npStore npDS.DataStore, query *v1.Query) ([]*v1.Namespace, error) {
-	metadataSlice, err := dataStore.SearchNamespaces(ctx, query)
+	metadataSlice, err := getNamespaceMetadataSliceFromQuery(ctx, dataStore, query)
 	if err != nil {
 		return nil, errors.Wrap(err, "retrieving namespaces")
 	}
 	return populateFromMetadataSlice(ctx, metadataSlice, deploymentDataStore, secretDataStore, npStore)
 }
 
+func getNamespaceMetadataSliceFromQuery(ctx context.Context, dataStore datastore.DataStore, query *v1.Query) ([]*storage.NamespaceMetadata, error) {
+	if loaders.HasLoaderContext(ctx) {
+		loader, err := loaders.GetNamespaceLoader(ctx)
+		if err != nil {
+			return nil, err
+		}
+		return loader.FromQuery(ctx, query)
+	}
+	return dataStore.SearchNamespaces(ctx, query)
+}
+
```

Comment:
I think it might be better to utilize the loader in the namespace graphql endpoint, and have that be the endpoint invoked by the service account graphql code